locator/abstract_replication_strategy.hh

```diff
@@ -98,6 +98,12 @@ public:
     virtual void validate_options() const = 0;
     virtual std::optional<std::set<sstring>> recognized_options() const = 0;
     virtual size_t get_replication_factor() const = 0;
+    // Decide if the replication strategy allow removing the node being
+    // replaced from the natural endpoints when a node is being replaced in the
+    // cluster. LocalStrategy is the not allowed to do so because it always
+    // returns the node itself as the natural_endpoints and the node will not
+    // appear in the pending_endpoints.
+    virtual bool allow_remove_node_being_replaced_from_natural_endpoints() const { return true; };
```

Comment:
If it is unsafe we should set it to false when we add new strategies.

Reply:
It is unsafe to have a default implementation. "false" is also not right for all strategies. This should be pure virtual so that each strategy has to decide for itself.

Reply:
If the strategy does not allow it, it should set to false. If it does not care, it should be the default which is true. I think it is overkill to make it pure virtual and ask every strategy to set it explicitly. It adds more code.

Reply:
Asias, I think Tomek is right on this one.

The goal is not to minimize code, it is to maximize safety. This is safe in the current code but as soon as you start adding more strategies, deriving from existing strategies, and end up forgetting to add this, then we just got ourselves a nasty bug.

Best to write a few extra lines of code now

Reply:
To have less code is not the goal but it is nice to have. It is trivial to change, I modified it in v6.